src/main/java/net/dv8tion/jda/bot/entities/impl/DefaultShardManagerImpl.java

```diff
@@ -0,0 +1,336 @@
+        JDAImpl jda = null;
+        try
+        {
+            if (this.queue.isEmpty())
+            {
+                jda = this.buildInstance(0);
+
+                this.shards = new ShardCacheViewImpl(this.shardsTotal);
+                for (int i = 0; i < this.shardsTotal; i++)
+                    this.queue.offer(i);
+
+                this.shards.getMap().put(0, jda);
+            }
+            else
+            {
+                final int shardId = this.queue.peek();
+
+                jda = this.buildInstance(shardId);
+                this.shards.getMap().put(shardId, jda);
+                this.queue.remove(shardId);
+            }
+        }
+        catch (final RateLimitedException e)
+        {
+            // do not remove 'shardId' from the queue and try the first one again after 5 seconds in the async thread
+        }
+        catch (final Exception e)
+        {
+            if (jda != null)
+                jda.shutdown();
+            throw e;
+        }
+
+        this.worker = this.executor.scheduleWithFixedDelay(() ->
+        {
+            if (this.queue.isEmpty())
+                return;
+
+            JDAImpl api = null;
+            try
+            {
+                int shardId = -1;
+                do
+                {
+                    final int i = this.queue.peek();
+                    if (this.shards.getMap().containsKey(i))
+                        this.queue.poll();
+                    else
+                        shardId = i;
+                }
+                while (shardId == -1 && !this.queue.isEmpty());
+
+                if (shardId == -1)
+                {
+                    this.queue.poll();
+                    return;
+                }
+
+                api = this.buildInstance(shardId);
+                this.shards.getMap().put(shardId, api);
+                this.queue.remove(shardId);
+            }
+            // as this happens before removing the shardId if from the queue just try again after 5 seconds
+            catch (RateLimitedException e)
+            {
+                LOG.warn("Hit the login ratelimit while creating new JDA instances, maybe try raising the login backoff");
+            }
+            catch (LoginException e)
+            {
+                // this can only happen if the token has been changed
+                // in this case the ShardManager will just shutdown itself as there currently is no way of hot-swapping the token on a running JDA instance.
+                LOG.warn("The token has been invalidated and the ShardManager will shutdown!");
+                this.shutdown();
+            }
+            catch (final Exception e)
+            {
+                if (api != null)
+                    api.shutdown();
+            }
+        }, 5000 + this.backoff, 5000 + this.backoff, TimeUnit.MILLISECONDS); // 5s for ratelimit + backoff for safety
```

Comment:
This `+ this.backoff` wouldn't be needed if you would wait for it to send the payload (waiting on JDA.Status to change to AWAITING_LOGIN_CONFIRMATION)